packages/chart/src/DonutSummary.ts

```diff
@@ -0,0 +1,79 @@
+import { Pie, SummaryC } from "@hpcc-js/chart";
+import { HTMLWidget } from "@hpcc-js/common";
+import { Layered } from "@hpcc-js/layout";
+
+export class DonutSummary extends HTMLWidget {
+    _layered;
+    _donut;
+    _summary;
+    constructor() {
+        super();
+        this._donut = new Pie();
+        this._summary = new SummaryC();
+    }
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        this._donut
+            .data(this.data())
+            .innerRadius(62)
+            ;
+        this._summary
+            .columns(this.summaryColumns())
+            .data(this.summaryData())
+            .iconColumn("Icon")
+            .labelColumn("Summary")
+            .valueColumn("Score")
+            .moreTextColumn("Details")
+            .moreTextHTML(true)
+            .colorFillColumn("Status")
+            .centeredValue(true)
+            .centeredLabel(true)
+            .iconOpacity(0)
+            ;
+        this._layered = new Layered()
+            .target(domNode)
+            .addLayer(this._donut)
+            .addLayer(this._summary, "center", 0.25, 0.25)
+            ;
+    }
+    summaryColumns() {
+        return ["Summary", "Score", "Details", "Status", "Icon"];
+    }
+    summaryData() {
+        return [
+            [
+                this.summaryText(),
+                this.summaryValue_exists() ? this.summaryValue() : this.data().reduce((acc, row) => acc + row[1], 0),
+                "",
+                "transparent",
+                "fa-info-circle"
+            ]
+        ];
+    }
+    update(domNode, element) {
+        super.update(domNode, element);
+        this._layered.resize({
+            height: this.height(),
+            width: this.width()
+        });
+        this._layered.render();
+    }
+    exit(domNode, element) {
+        super.exit(domNode, element);
+    }
+}
+DonutSummary.prototype._class += " chart_DonutSummary";
+
+export interface DonutSummary {
+    summaryText(): string;
+    summaryText(_: string): this;
+    summaryValue(): number;
+    summaryValue(_: number): this;
+    summaryValue_exists(): boolean;
+}
+DonutSummary.prototype.publish("summaryText", "", "string", "Subtext in centered summary widget");
+DonutSummary.prototype.publish("summaryValue", null, "number", "Value of the centered summary widget (defaults to sum of pie values when empty)");
+DonutSummary.prototype.publishProxy("paletteID", "_donut", "paletteID");
+DonutSummary.prototype.publishProxy("tooltipValueFormat", "_donut", "tooltipValueFormat");
+
+// TODO: Is there a simple way to expose all of the _donut publish properties here?
```

Comment:
I like the concept - but adding layout / layered and a Canvas widget is very heavy approach, when you could simply add two common/Text widgets.
Also with the Text widgets you can set their sizes and they will clip nicely with "..."s when needed.